src/main/capi/duckdb-c.cpp

```diff
@@ -129,6 +129,21 @@ duckdb_query_progress_type duckdb_query_progress(duckdb_connection connection) {
 	return query_progress_type;
 }
 
+void duckdb_set_progress_callback(duckdb_connection connection, duckdb_progress_callback_t callback) {
+	if (!connection) {
+		return;
+	}
+
+	Connection *conn = reinterpret_cast<Connection *>(connection);
+
+	if (!callback) {
```

Comment:
I don't see the point of this if statement, null is null, this branch is not needed, removing it would have the exact same effect.
I do see use in this if we take the wrapping approach, you probably don't want to wrap the callback if it's null anyways.

Reply:
The idea is to remove the callback if you are no longer interested in it.